contracts/optimistic-ethereum/OVM/OVM_BondManager.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.7.0;
+
+import { Lib_AddressResolver } from "../libraries/resolver/Lib_AddressResolver.sol";
+import { iOVM_FraudVerifier } from "../iOVM/verification/iOVM_FraudVerifier.sol";
+
+interface ERC20 {
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+}
+
+/// All the errors which may be encountered on the bond manager
+library Errors {
+    string constant ERC20_ERR = "BondManager: Could not post bond";
+    string constant NOT_ENOUGH_COLLATERAL = "BondManager: Sequencer is not sufficiently collateralized";
+    string constant LOW_VALUE = "BondManager: New collateral value must be greater than the previous one";
+    string constant ALREADY_FINALIZED = "BondManager: Fraud proof for this pre-state root has already been finalized";
+    string constant SLASHED = "BondManager: Cannot finalize withdrawal, you probably got slashed";
+    string constant CANNOT_CLAIM = "BondManager: Cannot claim yet. Dispute must be finalized first";
+
+    string constant WITHDRAWAL_PENDING = "BondManager: Withdrawal already pending";
+    string constant TOO_EARLY = "BondManager: Too early to finalize your withdrawal";
+
+    string constant ONLY_OWNER = "BondManager: Only the contract's owner can call this function";
+    string constant ONLY_TRANSITIONER = "BondManager: Only the transitioner for this pre-state root may call this function";
+    string constant ONLY_FRAUD_VERIFIER = "BondManager: Only the fraud verifier may call this function";
+    string constant ONLY_STATE_COMMITMENT_CHAIN = "BondManager: Only the state commitment chain may call this function";
+}
+
+contract OVM_BondManager is Lib_AddressResolver {
+    /// Owner used to bump the security bond size
+    address public owner;
+
+    /// The bond token
+    ERC20 immutable public token;
+
+    /// The fraud verifier contract, used to get data about transitioners for a pre-state root
+    // address public ovmFraudVerifier;
+    address public ovmCanonicalStateCommitmentChain;
+
+    uint256 requiredCollateral = 1 ether;
+
+    /// A bond posted by a sequencer
+    /// The bonds posted by each sequencer
+    struct Bond {
+        uint256 locked;
+        uint256 withdrawing;
+        uint256 withdrawalTimestamp;
+
+    }
+    mapping(address => Bond) public bonds;
+
+    /// The dispute period
+    uint256 public forceInclusionPeriodSeconds = 7 days;
```

Comment:
Totally our fault here -- `forceInclusionPeriodSeconds` is a variable in the `CanonicalTransactionChain` that should be renamed. It's meant to designate the period of time in which the sequencer can add transactions in advance of the `L1ToL2TransactionQueue`. We should probably change that variable inside the CTC to be more clear. 

Will want to rename `forceInclusionPeriodSeconds` here to `disputePeriodSeconds` or similar.